psutil/_psutil_posix.c

```diff
@@ -16,7 +16,7 @@
 #include <net/if.h>
 
 #ifdef PSUTIL_SUNOS10
-#include "arch/solaris10/ifaddrs.h"
+#include "arch/solaris/v10/ifaddrs.h"
```

Comment:
I can confirm that this fix is needed. I accidentally added this bug in e91f722 when fixing code review comments (file was moved, but include was not updated).
Installation failed on our Solaris 10 too, and I added the same fix in a local branch but did not push back to this project yet.
The problem is not with the architecture but with the version - Solaris v10 doesn't have ifaddrs.